.github/workflows/sonalcloud.yaml

```
@@ -0,0 +1,54 @@
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Needed for SonarCloud branch analysis
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@v3
+        with:
+          java-version: '21'
+          distribution: 'temurin'
+
+      - name: Cache Maven packages
+        uses: actions/cache@v3
```

actionlint (1.7.4)

the runner of "actions/cache@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)